spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroReaderForFieldsWithDefaultValue.java

```diff
@@ -50,72 +52,145 @@ public class TestSparkAvroReaderForFieldsWithDefaultValue {
 
   @Test
   public void testAvroDefaultValues() throws IOException {
-    String indexFiledName = "index";
-    String nullableFiledName = "optionalFieldWithDefault";
-    String requiredFiledName = "requiredFieldWithDefault";
-    int defaultValue = -1;
 
-    // write records with initial writeSchema
-    org.apache.avro.Schema writeSchema = org.apache.avro.Schema.createRecord("root", null, null, false,
-        ImmutableList.of(new org.apache.avro.Schema.Field(indexFiledName, org.apache.avro.Schema.create(INT),
-            null, null), new org.apache.avro.Schema.Field(nullableFiledName,
-            org.apache.avro.Schema.createUnion(org.apache.avro.Schema.create(INT),
-                org.apache.avro.Schema.create(NULL)), null, defaultValue)));
+    String writeSchemaString = "{\n" +
+            "  \"namespace\": \"com.n1\",\n" +
+            "  \"type\": \"record\",\n" +
+            "  \"name\": \"n1\",\n" +
+            "  \"fields\": [\n" +
+            "    {\n" +
+            "      \"name\": \"f1\",\n" +
```

Comment:
Can we use a field `f0` with string type (or whatever simple type) without default, and make this `f1` also testable in the subsequent read schema? The `string` primitive type should also be tested by this `f1` field.